display_list/dl_color.h

```diff
@@ -58,58 +77,99 @@ struct DlColor {
   static constexpr DlColor kCornflowerBlue()     {return DlColor(0xFF6495ED);};
   // clang-format on
 
-  constexpr bool isOpaque() const { return getAlpha() == 0xFF; }
-  constexpr bool isTransparent() const { return getAlpha() == 0; }
+  constexpr bool isOpaque() const { return alpha_ >= 1.f; }
+  constexpr bool isTransparent() const { return alpha_ <= 0.f; }
+
+  constexpr int getAlpha() const { return toC(alpha_); }
+  constexpr int getRed() const { return toC(red_); }
+  constexpr int getGreen() const { return toC(green_); }
+  constexpr int getBlue() const { return toC(blue_); }
 
-  constexpr int getAlpha() const { return argb_ >> 24; }
-  constexpr int getRed() const { return (argb_ >> 16) & 0xFF; }
-  constexpr int getGreen() const { return (argb_ >> 8) & 0xFF; }
-  constexpr int getBlue() const { return argb_ & 0xFF; }
+  constexpr DlScalar getAlphaF() const { return alpha_; }
+  constexpr DlScalar getRedF() const { return red_; }
+  constexpr DlScalar getGreenF() const { return green_; }
+  constexpr DlScalar getBlueF() const { return blue_; }
 
-  constexpr DlScalar getAlphaF() const { return toF(getAlpha()); }
-  constexpr DlScalar getRedF() const { return toF(getRed()); }
-  constexpr DlScalar getGreenF() const { return toF(getGreen()); }
-  constexpr DlScalar getBlueF() const { return toF(getBlue()); }
+  constexpr DlColorSpace getColorSpace() const { return color_space_; }
 
   constexpr uint32_t premultipliedArgb() const {
     if (isOpaque()) {
-      return argb_;
+      return argb();
     }
     DlScalar f = getAlphaF();
-    return (argb_ & 0xFF000000) |       //
+    return (argb() & 0xFF000000) |      //
```

Comment:
argb() converts the entire color only for you to extract just the alpha. Something more direct should work for the alpha value.

Also, what does this code do with float values out of range for the ColorSpaces that don't normalize to [0,1]?

Reply:
> argb() converts the entire color only for you to extract just the alpha. Something more direct should work for the alpha value.
> 
> Also, what does this code do with float values out of range for the ColorSpaces that don't normalize to [0,1]?

When everything is done the only thing using this will be the skia backend.  I think we should plan on this being deleted from dlcolor.  We have an integration test that shows impeller works fine with extended srgb.  It does that by never calling this.  I've added a deprecation notice.

Reply:
Does Skia use it? I just did a search in my workspace without these mods and the only use is from unit tests. I say, just delete it now...

Reply:
sgtm, removed.